From dc643585ecdf500693c82ee2645894b8ec6ab7c6 Mon Sep 17 00:00:00 2001
Message-Id: <dc643585ecdf500693c82ee2645894b8ec6ab7c6.1417780177.git.chang-joon.lee@intel.com>
In-Reply-To: <e293659a31a2970805251d9fed030ac809a9a6d0.1417780177.git.chang-joon.lee@intel.com>
References: <e293659a31a2970805251d9fed030ac809a9a6d0.1417780177.git.chang-joon.lee@intel.com>
From: Rodrigo Vivi <rodrigo.vivi@intel.com>
Date: Fri, 19 Sep 2014 20:16:26 -0400
Subject: [PATCH 11/30] REVERTME [VPG]: drm/i915: Add IS_BDW_GT3 macro.

It will be usefull to specify w/a that affects only BDW GT3.

v2: It is originally FROM_UPSTREAM but it uses INTEL_DEVID(dev) macro
which is not yet merged in our tree.
To fix compile errors and to continue with the rebase of other workaround
patches I have modified it to suit current convention hence it is a REVERTME.
Once we pull-in the upstream version this can be reverted.

For: VIZ-4092
Change-Id: I72b2c26500844a22e4ed1e5a3b4bea65b77f3486
Signed-off-by: Rodrigo Vivi <rodrigo.vivi@intel.com>
Upstream-repo: http://cgit.freedesktop.org/drm-intel/log/?h=drm-intel-next-queued
Upstream-commit: a0fcbd95354e5ac2400be74077cc668c91429ea7
Reviewed-by: Jesse Barnes <jbarnes@virtuousgeek.org>
Signed-off-by: Arun Siluvery <arun.siluvery@linux.intel.com>
---
 drivers/gpu/drm/i915/i915_drv.h |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_drv.h b/drivers/gpu/drm/i915/i915_drv.h
index cf4cf01..19f1d4d 100644
--- a/drivers/gpu/drm/i915/i915_drv.h
+++ b/drivers/gpu/drm/i915/i915_drv.h
@@ -2130,6 +2130,8 @@ struct drm_i915_cmd_table {
 				 (((dev)->pdev->device & 0xf) == 0x2  || \
 				 ((dev)->pdev->device & 0xf) == 0x6 || \
 				 ((dev)->pdev->device & 0xf) == 0xe))
+#define IS_BDW_GT3(dev)		(IS_BROADWELL(dev) && \
+				 ((dev)->pdev->device & 0x00F0) == 0x0020)
 #define IS_HSW_ULT(dev)		(IS_HASWELL(dev) && \
 				 ((dev)->pdev->device & 0xFF00) == 0x0A00)
 #define IS_ULT(dev)		(IS_HSW_ULT(dev) || IS_BDW_ULT(dev))
-- 
1.7.9.5

